via/static/scripts/video_player/utils/time.ts

```diff
@@ -0,0 +1,24 @@
+function pad(value: number): string {
+  return value.toString().padStart(2, '0');
+}
+
+/**
+ * Format a video timestamp.
+ *
+ * The timestamp is formatted as `m:ss` for times < 1 hour or `h:mm:ss` for
+ * times over 1 hour. This matches how transcript timestamps are formatted on
+ * youtube.com.
+ *
+ * @param time - Time in seconds since the start of the video
+ */
+export function formatTimestamp(time: number): string {
```

Comment:
This formatting is implemented manually for the moment without any consideration of the user's locale. I will need to check whether there are `Intl` methods we should be using instead.

Reply:
There is a spec proposal at https://github.com/tc39/proposal-intl-duration-format. See https://chromestatus.com/feature/5193021205774336 for browser engine links.